contracts/registry/src/lib.rs

```diff
@@ -885,4 +882,47 @@ mod tests {
         );
         assert_eq!(test_utils::get_logs(), vec![log_mint, log_renew].concat());
     }
+    #[test]
+    fn sbt_recover_basics() {
+        let (_, mut ctr) = setup(&issuer1(), 3 * MINT_DEPOSIT);
+        let m1_1 = mk_metadata(1, Some(START + 10));
+        let m2_1 = mk_metadata(2, Some(START + 10));
+        ctr.sbt_mint(vec![(alice(), vec![m1_1.clone(), m2_1.clone()])]);
+        assert_eq!(ctr.sbt_supply_by_owner(alice(), issuer1(), None), 2);
+
+        ctr.sbt_recover(alice(), bob());
+
+        let mint_log = mk_log_str(
+            "mint",
+            &format!(
+                r#"{{"issuer":"{}","tokens":[["{}",[1,2]]]}}"#,
+                issuer1(),
+                alice()
+            ),
+        );
+        let recover_log = mk_log_str(
+            "recover",
+            &format!(
+                r#"{{"issuer":"{}","old_owner":"{}","new_owner":"{}"}}"#,
+                issuer1(),
+                alice(),
+                bob()
+            ),
+        );
+        assert_eq!(test_utils::get_logs(), vec![mint_log, recover_log].concat());
```

Comment:
```suggestion
        let recover_log = mk_log_str(
            "recover",
            &format!(
                r#"{{"issuer":"{}","old_owner":"{}","new_owner":"{}"}}"#,
                issuer1(),
                alice(),
                bob()
            ),
        );
        assert_eq!(test_utils::get_logs().len(), 2);
        assert_eq!(test_utils::get_logs()[1], recover_log);
```

Reply:
Note, logs are "reset" (start from scratch) whenever we recreate test env with `testing_env!(...);`